lib/deep-equal.test.js

```diff
@@ -924,4 +924,38 @@ describe("deepEqual", function () {
             assert.isFalse(checkDeep);
         });
     });
+
+    describe("nested iterable", function () {
+        class Foo {
+            constructor(size = 0) {
+                // eslint-disable-next-line mocha/no-setup-in-describe
+                this.size = size;
+            }
+        }
+        // eslint-disable-next-line mocha/no-setup-in-describe
+        Foo.prototype[Symbol.iterator] = function () {
+            var count = 1;
+            // eslint-disable-next-line mocha/no-setup-in-describe
+            var size = this.size;
+            return {
+                next: function () {
+                    return count <= size
+                        ? { value: count++, done: false }
+                        : { value: undefined, done: true };
+                },
+            };
+        };
```

Comment:
This is what `before` is used for. Here's [an example](https://github.com/sinonjs/samsam/blob/3b2965042fe2d2f1e9e9989067ad0382f93eca8f/lib/deep-equal.test.js#L376-L385). Then you wouldn't have to disable mocha rules.

Reply:
@fearphage Thank you for your kind comment. I fixed it. https://github.com/sinonjs/samsam/pull/229/commits/4b7c73b93c6302b1e53d6aad3a9f8a173bc5bee0